### ISSUE TITLE
Docstring returned by analysis is over-escaped

### ISSUE THREAD

**Issue by lread** [clj-kondo/clj-kondo#1223]
**version**
2021.03.03

**platform**
All

**problem**
Clj-kondo analysis is not unescaping (or effectively, is over-escaping) docstrings returned in `:doc`.

**repro - as a library**
Given a file `repro.clj`, let's explore via an in editor REPL session:
```Clojure
(require '[clj-kondo.core :as clj-kondo])

(defn a-little-fn
  "A little function went to the \"store\""
  [a b]
  (+ a b))

;; return the docstring via Clojure meta on fn var
(-> #'a-little-fn
    meta
    :doc)
;; => "A little function went to the \"store\""

;; Return docstring via clj-kondo analysis, notice the extra escaping:
(-> (clj-kondo/run! {:lint ["fiddle.clj"]
                     :config {:output {:analysis true}}})
    :analysis
    :var-definitions
    first
    :doc)
;; => "A little function went to the \\\"store\\\""
```

**repro - from command line**

```Shell
> clj-kondo --lint repro.clj --config '{:output {:analysis true :format :edn}}'  \
     | jet --pretty --query ':analysis :var-definitions first :doc'
"A little function went to the \\\"store\\\""
```

**expected behavior**
Clj-kondo docstring should not be over-escaped; it should match the docstring returned by Clojure `:doc` meta on var.

**action** 
I'll take a stab at a PR.
